### ISSUE TITLE
tns v4.1.x: cannot find crypto module

### ISSUE THREAD

**Issue by jeffallen** [dedis/student_18_xplatform#35]
Cannot start it on Android because of:

```
System.err: java.lang.RuntimeException: Unable to start activity ComponentInfo{org.nativescript.cpmac/com.tns.NativeScriptActivity}: com.tns.NativeScriptException: Failed to find module: "crypto", relative to: app/tns_modules/
```

This is mentioned in: https://github.com/NativeScript/NativeScript/issues/195

A possible solution is: https://github.com/saulshanabrook/nativescript-webview-crypto

**Comment by yvgny** [comment]
Yes, sometimes this bug arises at launch time, but it's fixed by doing a 
`make clean-run-android`.
The crypto module is never used because it's replaced by crypto-browserify using the hook in nativescript-nodeify. I think this issue comes when the hook is not correctly launched by NativeScript CLI.
Could you try this on your side ?
Thank you

**Comment by jeffallen** [comment]
Happens every time for me.
```
icsil1noteb238:cpmac jallen$ tns --version
4.1.0
icsil1noteb238:cpmac jallen$ node --version
v8.11.2
```

**Comment by yvgny** [comment]
Apparently it's due to the nativescript-nodeify not correctly launching the after-prepare hook on 4.1 version : https://github.com/EddyVerbruggen/nativescript-nodeify/issues/49 . NativeScript 4.1 is out since 9 days, so I didn't notice this issue. For the moment I think it's safer to use versions 3.x.x until nativescript-nodeify gets patched. I'll try to see if I can patch it as soon as I have some time.

**Comment by jeffallen** [comment]
Issue 49 is the problem. Confirmed that this works around the problem:
```
npm uninstall -g nativescript
npm install -g nativescript@4.0.2
```

**Comment by ineiti** [comment]
Needs to be fixed to work on the latest nativescript version.

**Comment by jeffallen** [comment]
Fixed in ad4f56f65b.
